tests/console/openvpn.pm

```diff
@@ -0,0 +1,65 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Setup and test point-to-point VPN using static key with OpenVPN
+# Maintainer: Dehai Kong <dhkong@suse.com>
+
+use base "consoletest";
+use strict;
+use testapi;
+use utils;
+use lockapi;
+use mmapi;
+use mm_network;
+use mm_tests;
+
+sub run {
+
+    my $password = $testapi::password;
+    select_console "root-console";
+
+    # Set up openvpn client's IP address
+    configure_static_network('10.0.2.10/24');
+
+    # Stop firewall
+    assert_script_run("SuSEfirewall2 stop");
+
+    mutex_lock('openvpn');
+    mutex_unlock('openvpn');
+
+    # Install openvpn and get static key from openvpn server
+    zypper_call("in openvpn");
+    assert_script_run("mkdir -p /etc/openvpn/keys");
+    script_run("scp root\@10.0.2.1:/etc/openvpn/keys/openvpn.key /etc/openvpn/keys/ | tee /dev/$serialdev", 0);
+    wait_still_screen 1;
+    type_string("yes\n");
+    wait_still_screen 1;
+    type_string("$password\n");
+    assert_script_run("ls /etc/openvpn/keys/openvpn.key");
+
+    # Setup VPN client
+    my $client_conf = <<EOF;
```

Comment:
you duplicate this config in two places . Better move it to data folder and then extract using curl and autoinst_url . Here is example https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/slepos/register_images.pm#L24

Reply:
well, the config is not really same. It only shares some parts. While in general I fully support to reduce duplication IMHO here it would be too much effort but still possible. *If* you still want to extract it then I suggest to define only the server config and generate the client config on the fly by reading the server config and replacing parts of it and adding the line `remote 10.0.2.1` on top.

Reply:
actually, there are two different lines between server and client configuration. some part of this case, I will try to compact into one config sub. to let it more lightweight.